integration-tests/src/test/scala/org/constellation/TransactionsProcessingSpec.scala

```diff
@@ -0,0 +1,144 @@
+package org.constellation
+
+import java.security.KeyPair
+
+import cats.effect.{ContextShift, IO, Timer}
+import cats.syntax.all._
+import com.typesafe.scalalogging.StrictLogging
+import fs2.{Pipe, Stream}
+import io.circe.syntax._
+import org.constellation.keytool.KeyStoreUtils
+import org.constellation.schema.address.AddressCacheData
+import org.constellation.schema.transaction.{LastTransactionRef, Transaction}
+import org.constellation.wallet.{TransactionExt, Wallet}
+import org.http4s.circe.CirceEntityDecoder._
+import org.http4s.circe.CirceEntityEncoder._
+import org.http4s.client.blaze.BlazeClientBuilder
+import org.http4s.client.dsl.io._
+import org.http4s.dsl.io._
+import org.http4s.{EntityDecoder, Request, Uri}
+import org.mockito.IdiomaticMockito
+import org.scalatest.GivenWhenThen
+import org.scalatest.concurrent.Eventually
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.matchers.{MatchResult, Matcher}
+import org.scalatest.time.{Minutes, Seconds, Span}
+
+import scala.concurrent.ExecutionContext.global
+
+class TransactionsProcessingSpec
+    extends AnyFreeSpec
+    with StrictLogging
+    with GivenWhenThen
+    with Matchers
+    with TestConfig
+    with TerraformOutput
+    with Eventually
+    with IdiomaticMockito {
+
+  implicit val cs: ContextShift[IO] = IO.contextShift(global)
+  implicit val timer: Timer[IO] = IO.timer(global)
+
+  override implicit val patienceConfig: PatienceConfig =
+    PatienceConfig(
+      timeout = scaled(Span(2, Minutes)),
+      interval = scaled(Span(5, Seconds))
+    )
+
+  private val containTheSameElements = (not be empty).and(Matcher { (left: Seq[_]) =>
+    MatchResult(
+      left.forall(_ == left.head),
+      left + " did not contain the same elements",
+      left + " did contain the same elements"
+    )
+  })
+
+  "sending transactions from single src address" in {
+    // given
+    val kp = getKeyPair
+    val (src, dst) = (Wallet.getAddress(kp), config.txGen.dst)
+
+    val srcBalances = getBalances(src)
+    srcBalances should containTheSameElements
+
+    val dstBalances = getBalances(dst)
+    dstBalances should containTheSameElements
+    logger.info(s"Initial balances: src address = $src, src balance = ${srcBalances.head}; dst address = $dst, dst balance = ${dstBalances.head}")
+
+    val lastTxRefs = getLasTxRefs(src)
+    lastTxRefs should containTheSameElements
+
+    // when
+    val txs = generateAndSendTxs(kp, lastTxRefs.head)
+    logger.info(s"Sent transactions $txs")
+
+    // then
+    val targetSrcBalance = srcBalances.head - (config.txGen.amount + config.txGen.fee) * config.txGen.countTotal
+    val targetDstBalance = dstBalances.head + config.txGen.amount * config.txGen.countTotal
+
+    eventually {
+      getBalances(src) should contain only targetSrcBalance
+      getBalances(dst) should contain only targetDstBalance
+    }
+  }
+
+  private def getKeyPair =
```

Comment:
I wonder if we really need to load up the key from a file? I guess for this test we could just as well just create a random keyPair with keytool. Just food for thought if we really need to load the keys.

Reply:
Probably other more complicated tests will need that.

Reply:
We need these keys as these addresses are pre-filled in genesis. But we shouldn't fetch all of them, that's for sure

Reply:
Right, forgot about genesis, and that we need some positive balance.

Reply:
I agree with @TheMMaciek ideally we should generate these keys on the fly as well as genesis that corresponds with those keys. I didn't do that with this change, but I think it wouldn't hurt to have a task for it.

This is an open-source project, and if someone else would like to contribute to it, he wouldn't be able to run these tests without access to our S3 bucket.

Reply:
@tbekas Everybody is able to run own cluster, there is no need to have an access to our S3.

Reply:
yet, you cannot use autogenerated keys

Reply:
Those keys aren't public, aren't they? So any contributor would have to generate them on his own, as well as genesis. Lot's of unnecessary manual work, that should be automated.